src/neural_network/layer/neuron/SimpleNeuron.cpp

```diff
@@ -15,19 +15,21 @@ SimpleNeuron::SimpleNeuron(NeuronModel model, shared_ptr<NeuralNetworkOptimizer>
 float SimpleNeuron::output(const vector<float>& inputs)
 {
     this->lastInputs = inputs;
-    this->sum = 0;
+    float tmp = 0.0f; // to activate the SIMD optimization
+    #pragma omp simd
     for (size_t w = 0; w < this->weights.size(); ++w)
     {
-        this->sum += inputs[w] * weights[w];
+        tmp += inputs[w] * weights[w];
     }
-    this->sum += bias;
+    this->sum = tmp + bias;
     return this->outputFunction->function(this->sum);
 }
 
 vector<float>& SimpleNeuron::backOutput(float error)
 {
     error = error * this->outputFunction->derivative(this->sum);
 
+    #pragma omp simd // seems to do nothing
```

Comment:
Replace `#pragma omp simd` with `#pragma omp parallel for num_threads(/*number of logical cores*/) schedule(runtime)`

Reply:
I have to think if I want to set multi-core in this branch, if so I have to think about how to configure the number of cores to use in the library.